library/core/src/hint.rs

```diff
@@ -96,23 +96,25 @@ pub fn spin_loop() {
     }
 }
 
-/// An identity function that *__hints__* to the compiler to be maximally pessimistic about what
-/// `black_box` could do.
+/// An identity function that suggests to the compiler that the given value is used in some
+/// arbitrary, unspecified way.
```

Comment:
This wording seems really confusing to me, because this can imply two incorrect interpretations:
1. It is used by the caller of `assume_used`, but that's already obvious, how else are we going to call the function without using the value we pass in? So that interpretation is entirely meaningless.
2. It is used by `assume_used` in an unspecified way. This however implies that instead of preventing optimizations, the compiler is free to return any arbitrary value as this seemingly isn't required to return the value back to me. So this interpretation would actually imply the opposite: The compiler is encouraged to optimize my code away.